pkg/httpproxy/proxy.go

```diff
@@ -230,16 +234,27 @@ func (p *proxy) secretGetter(req *http.Request, cAuth string) SecretGetter {
 }
 
 func (p *proxy) checkCluster(req *http.Request, user user.Info, clusterID, credID string) (authorizer.Decision, error) {
-	cluster, err := p.clusters.Controller().Lister().Get("", clusterID)
+	cluster, err := p.mgmtClusters.Controller().Lister().Get("", clusterID)
 	if err != nil {
 		return authorizer.DecisionDeny, err
 	}
-	if cluster.Spec.EKSConfig == nil {
+
+	administratedByProvV2Cluster := imported.IsAdministratedByProvisioningCluster(cluster)
+	if (cluster.Spec.EKSConfig == nil || cluster.Spec.EKSConfig.AmazonCredentialSecret != credID) && !administratedByProvV2Cluster {
 		return authorizer.DecisionDeny, nil
 	}
-	if cluster.Spec.EKSConfig.AmazonCredentialSecret != credID {
-		return authorizer.DecisionDeny, nil
+
+	if administratedByProvV2Cluster {
+		provCluster, err := p.provClusters.Get(cluster.Annotations["objectset.rio.cattle.io/owner-namespace"], cluster.Annotations["objectset.rio.cattle.io/owner-name"])
+		if err != nil {
+			return authorizer.DecisionDeny, err
+		}
+
+		if provCluster.Status.ClusterName != clusterID || provCluster.Spec.CloudCredentialSecretName != credID {
```

Comment:
Is there any validation to prevent someone from adding any credential they like to Spec.CloudCredentialSecretName or changing the Status.ClusterName? If status is a block subschema that should be fine too. ClusterName may be fine if them being able to edit provisioning cluster means they are an owner, similar to norman clusters since to do anything with this they'd need to have access to both clusters anyway.

Reply:
1. Status is a sub-object of the provisioning cluster.
2. There is logic [in the webhook](https://github.com/rancher/webhook/blob/master/pkg/resources/validation/cluster/provisioningcluster.go#L67) that validates that the user has access to the cloud credential they are trying to change to.